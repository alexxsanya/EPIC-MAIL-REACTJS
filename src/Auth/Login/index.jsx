import React from 'react';
import {
  Form,
  Row,
  Col,
  Button,
} from 'react-bootstrap';
import { connect } from 'react-redux';
import Loader from 'react-loader';
import '../CSS/Auth.scss';
import loginUser from '../Redux/actions/loginAction';
import StatusLabel from '../../common/components/StatusLabel';

class Login extends React.Component {
  constructor (props){
    super(props);
    this.state = {
      email: '',
      password: '',
    }
  }

  componentWillMount() {
    const token = localStorage.getItem('user_token');
    if(token !== null ) {
      window.location = '/dashboard/inbox'
    }
  }

  handleChange = event => {
    const {name, value} = event.target;
    this.setState({
      [name]: value,
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.setState({
      disabled: true
    })
    const { email, password} = this.state
    const userData = {
        email: email,
        password: password,
    }
    
    this.props.loginUser(userData);
  }

  render() {
    const { password, email, disabled } = this.state;
    const { isLoggingIn, errorOccured, isLoggedIn, response,userInfo } = this.props

    if (userInfo.status === 200) {
      localStorage.setItem("user_token", userInfo.data[0].token);
      setTimeout(() => {
        window.location = `/dashboard/inbox`;
      },
      2000);
    }

    return (
      <div className="login-component">
        <h2 className="section-title">Login Here</h2>
        <Row className="status-section">
          <Col sm={12}>
              {
                errorOccured.errorMessage === undefined
                && (
                <StatusLabel
                message={String(response)}
                tip="Check Your Network & Try Again"
                />
                )
              }
              {
                (userInfo.data === [] || userInfo.status === 400)
                && (
                <StatusLabel
                message="No user with provided credentials"
                tip="try again"
                />
                )
              }
              {
                (userInfo.status === 200)
                && <StatusLabel 
                message="You have been logged in successfully" 
                tip=" "
                variant="success"
                />
              }
          </Col>
        </Row>
        <Form onSubmit={this.handleSubmit}>
          <Form.Group controlId="formBasicTitle">
            <Form.Control
              type="email"
              required
              name="email"
              placeholder="Enter your email"
              value={email}
              onChange={this.handleChange}
            />
          </Form.Group>
          <Form.Group controlId="formBasicTitle">
            <Form.Control
              type="password"
              required
              name="password"
              placeholder="Enter your password"
              value={password}
              onChange={this.handleChange}
            />
          </Form.Group>
          <Row>
            <Col sm={12} className="call-for-action">
              <Loader loaded={!isLoggingIn} color="#3379f5">
                <Button type="submit" id="submitForm" disabled={isLoggedIn}>
                    Login
                </Button>
              </Loader>
            </Col>
          </Row>
          <Row>
            <Col sm={12} className="label-action">
              <span>Create an account, click <a href='/signup'>here</a> to signup</span>
            </Col>
          </Row>
        </Form>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
    isLoggingIn: state.loginReducer.isLoggingIn,
    errorOccured: state.loginReducer.error,
    isLoggedIn: state.loginReducer.status,
    userInfo: state.loginReducer.item,
    response: state.loginReducer.message
  });
  
export default connect(mapStateToProps, { loginUser })(Login);