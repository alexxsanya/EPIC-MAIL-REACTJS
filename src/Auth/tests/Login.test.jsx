/* eslint-disable no-undef */
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Login } from '../Login';

const props = {
  isLoggingIn: false,
  errorOccured: {
    errorMessage: 'No Message',
  },
  isLoggedIn: false,
  userInfo: {},
  response: 'Login Failed',
};

describe('Login Component', () => {
  it('should match the snapshot', () => {
    const component = shallow(<Login {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call handleSubmit upon login ', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const handleSubmitMock = jest.fn(event);
    const wrapper = mount(<Login {...props} />);

    expect(wrapper).toBeDefined();
    // expect(handleSubmitMock).toBeCalled();
  });
});
