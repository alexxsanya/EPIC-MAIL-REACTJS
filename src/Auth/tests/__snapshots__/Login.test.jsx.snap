// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match the snapshot 1`] = `
<div
  className="login-component"
>
  <h2
    className="section-title"
  >
    Login Here
  </h2>
  <ForwardRef
    className="status-section"
    noGutters={false}
  >
    <Col
      sm={12}
    />
  </ForwardRef>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <FormGroup
      controlId="formBasicTitle"
    >
      <FormControl
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        required={true}
        type="email"
        value=""
      />
    </FormGroup>
    <FormGroup
      controlId="formBasicPass"
    >
      <FormControl
        name="password"
        onChange={[Function]}
        placeholder="Enter your password"
        required={true}
        type="password"
        value=""
      />
    </FormGroup>
    <ForwardRef
      noGutters={false}
    >
      <Col
        className="call-for-action"
        sm={12}
      >
        <Component
          color="#3379f5"
          component="div"
          loaded={true}
          loadedClassName="loadedContent"
          parentClassName="loader"
        >
          <Button
            active={false}
            disabled={false}
            id="submitForm"
            type="submit"
            variant="primary"
          >
            Login
          </Button>
        </Component>
      </Col>
    </ForwardRef>
    <ForwardRef
      noGutters={false}
    >
      <Col
        className="label-action"
        sm={12}
      >
        <span>
          Create an account, click 
          <a
            href="/signup"
          >
            here
          </a>
           to signup
        </span>
      </Col>
    </ForwardRef>
  </Form>
</div>
`;
